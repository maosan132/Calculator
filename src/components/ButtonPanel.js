import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = (props) => {
  const firstRow = ['AC', '+/-', '%', '/'];
  const secondRow = ['7', '8', '9', 'X'];
  const thirdRow = ['4', '5', '6', '-'];
  const fourthRow = ['1', '2', '3', '+'];
  const fifthRow = ['0', '.', '='];

  const buttonFactory = (row) => {
    const x = row.map((item, index) => (
      <Button
        key={row[index]}
        handleClick={props.clickHandler(item)}
        name={item}
        wide={item === '0'}
      />
    ));
    return x;
  };

  return (
    <div className="ButtonPanel">
      <div className="row1">
        {buttonFactory(firstRow)}
      </div>
      <div className="row2">
        {buttonFactory(secondRow)}
      </div>
      <div className="row3">
        {buttonFactory(thirdRow)}
      </div>
      <div className="row4">
        {buttonFactory(fourthRow)}
      </div>
      <div className="row5">
        {buttonFactory(fifthRow)}
      </div>
    </div>
  );
};

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func,
};

ButtonPanel.defaultProps = {
  clickHandler: null,
};

export default ButtonPanel;
