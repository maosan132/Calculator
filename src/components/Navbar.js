import React from 'react';
import { Link } from 'react-router-dom';

const NavBar = () => (
  <div id="navbar">
    <h1>Math Magicians</h1>
    <div className="nav">
      <Link to="/io/math-magicians/">Home </Link>
      <Link to="/io/math-magicians/App">Calculator </Link>
      <Link to="/io/math-magicians/Quote">Quote </Link>
    </div>
  </div>
);

export default NavBar;
