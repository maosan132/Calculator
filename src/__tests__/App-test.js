import React from 'react';
import { BrowserRouter as Router} from 'react-router-dom';
import Renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';
import Display from '../components/Display';
import ButtonPanel from '../components/ButtonPanel';

describe('App', () => {
  it('renders correctly', () => {
    const app = Renderer
      .create(
        <div className="app">
          <Router>
            <Navbar />
          </Router>
          <Display />
          <ButtonPanel />
        </div>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
