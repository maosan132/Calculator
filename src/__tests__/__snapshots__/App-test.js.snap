// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="app"
>
  <div
    id="navbar"
  >
    <h1>
      Math Magicians
    </h1>
    <div
      className="nav"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        href="/App"
        onClick={[Function]}
      >
        Calculator 
      </a>
      <a
        href="/Quote"
        onClick={[Function]}
      >
        Quote 
      </a>
    </div>
  </div>
  <div
    className="Display"
  >
    0
  </div>
  <div
    className="ButtonPanel"
  >
    <div
      className="row1"
    >
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="%"
      >
        %
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#F5913E",
          }
        }
        type="button"
        value="/"
      >
        /
      </button>
    </div>
    <div
      className="row2"
    >
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="7"
      >
        7
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="8"
      >
        8
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="9"
      >
        9
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#F5913E",
          }
        }
        type="button"
        value="X"
      >
        X
      </button>
    </div>
    <div
      className="row3"
    >
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="4"
      >
        4
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="5"
      >
        5
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="6"
      >
        6
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#F5913E",
          }
        }
        type="button"
        value="-"
      >
        -
      </button>
    </div>
    <div
      className="row4"
    >
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="1"
      >
        1
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="2"
      >
        2
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="3"
      >
        3
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#F5913E",
          }
        }
        type="button"
        value="+"
      >
        +
      </button>
    </div>
    <div
      className="row5"
    >
      <button
        className="zero-button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="0"
      >
        0
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "initial",
          }
        }
        type="button"
        value="."
      >
        .
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#F5913E",
          }
        }
        type="button"
        value="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;
